protocol/messenger_walletconnect.go

```diff
@@ -30,6 +32,28 @@ func (m *Messenger) AddWalletConnectSession(request *requests.AddWalletConnectSe
 	return m.persistence.InsertWalletConnectSession(session)
 }
 
+func (m *Messenger) NewWalletConnectV2SessionCreatedNotification(session walletconnect.Session) error {
+	now := m.GetCurrentTimeInMillis()
+
+	notification := &ActivityCenterNotification{
+		ID:                         types.FromHex(string(session.Topic) + "_dapp_connected"),
+		Type:                       ActivityCenterNotificationTypeDAppConnected,
+		DAppURL:                    session.Peer.Metadata.URL,
+		DAppName:                   session.Peer.Metadata.Name,
+		WalletProviderSessionTopic: string(session.Topic),
+		Timestamp:                  now,
+		UpdatedAt:                  now,
+	}
+
+	if len(session.Peer.Metadata.Icons) > 0 {
+		notification.DAppIconURL = session.Peer.Metadata.Icons[0]
+	}
+
+	_, err := m.persistence.SaveActivityCenterNotification(notification, true)
+
```

Comment:
To notify client here, you'd do something like this:
```go
if m.config.messengerSignalsHandler != nil {
    response := &MessengerResponse{}
    response.AddActivityCenterNotification(notification)
    m.config.messengerSignalsHandler.MessengerResponse(response)
}
```

Reply:
Though combine it with my next comment.

Reply:
thanks, i added and now works fine.